### PR TITLE
display staking cap before wallet login

### PR DESCRIPTION
**Pull request by jrwbabylonchain**
image

Ideally the global param and height fetch in the page.tsx can be improved using same/similar method.
But since we are currently at a testing stage. we can leave it for now and deal with it later on

### CODE REVIEW

File: src/app/context/api/GlobalParamsProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { ReactNode, createContext, useContext } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { getTipHeight } from "@/utils/mempool_api";
```
Comment (gbarkhatov) [inline src/app/context/api/GlobalParamsProvider.tsx:3]: should we separate package imports from the local imports?

File: src/app/context/api/GlobalParamsProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+const GlobalParamsContext = createContext<GlobalParamsVersion[] | undefined>(
+  undefined,
+);
+
+export const GlobalParamsProvider: React.FC<GlobalParamsProviderProps> = ({
+  children,
+}) => {
+  const { data } = useQuery({
+    queryKey: ["API_GLOBAL_PARAMS"],
```
Comment (gbarkhatov) [inline src/app/context/api/GlobalParamsProvider.tsx:19]: I think we need to either uppercase all of them, or lowercase all of them:
https://i.imgur.com/fo9t50R.png

**Review by gbarkhatov — APPROVED** [review]

File: src/app/context/api/GlobalParamsProvider.tsx (hunk 2 above)
Comment (jrwbabylonchain) [reply src/app/context/api/GlobalParamsProvider.tsx:19]: yeah. generally speaking, i prefer uppercase and no space (use `_`) for any key identifier.
If you agree with this approach, we can have a low priority task to change all existing ones to this.
But again, i don't have strong opinion on this. If not bothered, happy to change back to the current format which i think is `api global params`

File: src/app/context/api/GlobalParamsProvider.tsx (hunk 1 above)
Comment (jrwbabylonchain) [reply src/app/context/api/GlobalParamsProvider.tsx:3]: probably a good way to handle this to let the npx format to fix it.

**Pull request merged**